File: frontend/src/types/violation.ts
export type ViolationType =
  | 'helmet_violation'
  | 'signal_jump'
  | 'wrong_way'
  | 'speeding'
  | 'no_seatbelt'
  | 'illegal_parking';

export type ViolationStatus =
  | 'detected'
  | 'confirmed'
  | 'rejected'
  | 'evidence_generated'
  | 'sent_to_authority';

export interface Violation {
  id: string;
  camera_id: string;
  violation_type: ViolationType;
  status: ViolationStatus;
  license_plate: string | null;
  confidence: number;
  clip_url: string | null;
  thumbnail_url: string | null;
  evidence_package_url: string | null;
  location: string | null;
  detected_at: string;
  created_at: string;
}

export interface ViolationStats {
  total_violations: number;
  by_type: Record<string, number>;
  by_status: Record<string, number>;
  by_camera: Record<string, number>;
  today_count: number;
  this_week_count: number;
}

export interface Camera {
  id: string;
  name: string;
  stream_url: string;
  location: string | null;
  status: string;
  created_at: string;
}

File: backend/app/models/violation.py
import uuid
from datetime import datetime
from enum import Enum as PyEnum

from sqlalchemy import Column, DateTime, Enum, Float, String, Text
from sqlalchemy.dialects.postgresql import UUID

from backend.app.core.database import Base


class ViolationType(str, PyEnum):
    HELMET_VIOLATION = "helmet_violation"
    SIGNAL_JUMP = "signal_jump"
    WRONG_WAY = "wrong_way"
    SPEEDING = "speeding"
    NO_SEATBELT = "no_seatbelt"
    ILLEGAL_PARKING = "illegal_parking"


class ViolationStatus(str, PyEnum):
    DETECTED = "detected"
    CONFIRMED = "confirmed"
    REJECTED = "rejected"
    EVIDENCE_GENERATED = "evidence_generated"
    SENT_TO_AUTHORITY = "sent_to_authority"


class Violation(Base):
    __tablename__ = "violations"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    camera_id = Column(String(100), nullable=False, index=True)
    violation_type = Column(Enum(ViolationType), nullable=False, index=True)
    status = Column(Enum(ViolationStatus), default=ViolationStatus.DETECTED, index=True)
    license_plate = Column(String(20), index=True)
    confidence = Column(Float, nullable=False)
    clip_url = Column(Text)
    thumbnail_url = Column(Text)
    evidence_package_url = Column(Text)
    location = Column(String(255))
    detected_at = Column(DateTime, default=datetime.utcnow, index=True)
    created_at = Column(DateTime, default=datetime.utcnow)
    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)
    metadata_json = Column(Text)


class Camera(Base):
    __tablename__ = "cameras"

    id = Column(String(100), primary_key=True)
    name = Column(String(255), nullable=False)
    stream_url = Column(Text, nullable=False)
    location = Column(String(255))
    status = Column(String(50), default="active")
    created_at = Column(DateTime, default=datetime.utcnow)
